### DISCUSSION TITLE
possible to store client args with auth user?

### DISCUSSION THREAD

**Discussion by Alex-Wenner-FHR** [h2oai/wave#924] [Q&A] [answered]
Is it possible for client args to be stored with an auth user? For example, a user can customize a table, select/deselect columns. this data is stored in something like q.client.table. Is it possible to provide this data each time the associated q.auth.subject logs onto the app?

**Comment by lo5** [comment, marked as answer]
User info is best stored in `q.user`, not `q.client`. 
`q.user` is tied to a user, whereas `q.client` is tied to a browser tab.

If you store info in `q.user`, you can use [checkpointing](https://wave.h2o.ai/docs/configuration#h2o_wave_checkpoint_dir) to restore that info between app restarts.

**Reply by Alex-Wenner-FHR** [reply]
Ah yes can't believe I didn't think of that! Thanks!
